cumulusci/tasks/package_2gp.py

```diff
@@ -0,0 +1,537 @@
+            if dependency.get("version_id"):
+                name = (
+                    f"{dependency['namespace']}@{dependency['version']} "
+                    if "namespace" in dependency
+                    else ""
+                )
+                self.logger.info(
+                    f"Adding dependency {name}with id {dependency['version_id']}"
+                )
+
+            elif dependency.get("repo_name"):
+                version_id = self._create_unlocked_package_from_github(dependency)
+                self.logger.info(
+                    "Adding dependency {}/{} {} with id {}".format(
+                        dependency["repo_owner"],
+                        dependency["repo_name"],
+                        dependency["subfolder"],
+                        version_id,
+                    )
+                )
+                new_dependency["subscriberPackageVersionId"] = version_id
+
+            else:
+                raise DependencyLookupError(
+                    f"Unable to convert dependency: {dependency}"
+                )
+
+            new_dependencies.append(new_dependency)
+
+        return new_dependencies
+
+    def _get_unpackaged_pre_dependencies(self, dependencies):
+        """Create package for unpackaged/pre metadata, if necessary
+        """
+        path = pathlib.Path("unpackaged", "pre")
+        if not path.exists():
+            return dependencies
+
+        for item_path in sorted(path.iterdir(), key=str):
+            if item_path.is_dir():
+                version_id = self._create_unlocked_package_from_local(item_path)
+                self.logger.info(
+                    "Adding dependency {}/{} {} with id {}".format(
+                        self.project_config.repo_owner,
+                        self.project_config.repo_name,
+                        item_path,
+                        version_id,
+                    )
+                )
+                dependencies.append({"subscriberPackageVersionId": version_id})
+
+        return dependencies
+
+    def _create_unlocked_package_from_github(self, dependency):
+        gh_for_repo = self.project_config.get_github_api(
+            dependency["repo_owner"], dependency["repo_name"]
+        )
+        zip_src = download_extract_github(
+            gh_for_repo,
+            dependency["repo_owner"],
+            dependency["repo_name"],
+            dependency["subfolder"],
+            ref=dependency.get("ref"),
+        )
+        package_zip_builder = MetadataPackageZipBuilder.from_zipfile(
+            zip_src, options=dependency, logger=self.logger
+        )
+
+        package_config = PackageConfig(
+            name="{repo_owner}/{repo_name} {subfolder}".format(**dependency),
+            version_name="Auto",
+            package_type="Unlocked",
+            # Ideally we'd do this without a namespace,
+            # but it needs to match the dependent package
+            namespace=self.package_config.namespace,
```

Comment:
This _shouldn't_ break package builds (although it's possible if someone did something a bit foolish and created dependencies between unpackaged/pre metadata in different packages), but I think there's a good chance that we'll have some broken setup automation. For example, packages downstream of NPSP that do further customization may depend on the unpackaged Record Types not being namespaced. `load_dataset` is _probably_ resilient to that because of how it resolves record type names, but other automation may not be.

What's the source of the need for the namespaces to match?

Reply:
Yeah, it's not ideal. If you try to use an un-namespaced package as a dependency of another 2gp package, you get an error saying you can't do that. It doesn't need to match but it needs to have a namespace, and using the same one as the primary package being created is simplest because we know it's already in the devhub's namespace registry.